src/Simplex/Chat/Store.hs

```diff
@@ -4107,88 +4094,62 @@ setChatItemTTL db User {userId} chatItemTTL = do
-        FROM chat_items
-        WHERE user_id = ? AND item_ts <= ?
-        GROUP BY contact_id, group_id
-        ORDER BY contact_id ASC, group_id ASC
-      |]
-      (userId, expirationDate)
-  where
-    toChatRef :: (Maybe ContactId, Maybe GroupId) -> Maybe ChatRef
-    toChatRef (Just contactId, Nothing) = Just $ ChatRef CTDirect contactId
-    toChatRef (Nothing, Just groupId) = Just $ ChatRef CTGroup groupId
-    toChatRef _ = Nothing
-
-getContactExpiredCIs :: DB.Connection -> User -> ContactId -> UTCTime -> IO [(ChatItemId, Maybe CIFileInfo)]
-getContactExpiredCIs db User {userId} contactId expirationDate =
-  map toItemIdAndFileInfo'
-    <$> DB.query
-      db
-      [sql|
-        SELECT i.chat_item_id, f.file_id, f.ci_file_status, f.file_path
-        FROM chat_items i
-        LEFT JOIN files f ON f.chat_item_id = i.chat_item_id
-        WHERE i.user_id = ? AND i.contact_id = ? AND i.item_ts <= ?
-        ORDER BY i.item_ts ASC
-      |]
-      (userId, contactId, expirationDate)
+getContactCICount :: DB.Connection -> User -> Contact -> IO (Maybe Int64)
+getContactCICount db User {userId} Contact {contactId} =
+  fmap join . maybeFirstRow fromOnly $
+    DB.query db "SELECT COUNT(1) FROM chat_items WHERE user_id = ? AND contact_id = ?" (userId, contactId)
 
-getGroupExpiredCIs :: DB.Connection -> User -> Int64 -> UTCTime -> IO [(ChatItemId, Maybe CIFileInfo)]
-getGroupExpiredCIs db User {userId} groupId expirationDate =
-  map toItemIdAndFileInfo'
+getGroupExpiredFileInfo :: DB.Connection -> User -> GroupInfo -> UTCTime -> UTCTime -> IO [CIFileInfo]
+getGroupExpiredFileInfo db User {userId} GroupInfo {groupId} expirationDate createdAtCutoff =
+  map toFileInfo
     <$> DB.query
       db
       [sql|
-        SELECT i.chat_item_id, f.file_id, f.ci_file_status, f.file_path
+        SELECT f.file_id, f.ci_file_status, f.file_path
         FROM chat_items i
-        LEFT JOIN files f ON f.chat_item_id = i.chat_item_id
-        WHERE i.user_id = ? AND i.group_id = ? AND i.item_ts <= ?
-        ORDER BY i.item_ts ASC
+        JOIN files f ON f.chat_item_id = i.chat_item_id
+        WHERE i.user_id = ? AND i.group_id = ? AND i.item_ts <= ? AND i.created_at <= ?
       |]
-      (userId, groupId, expirationDate)
+      (userId, groupId, expirationDate, createdAtCutoff)
 
-toItemIdAndFileInfo' :: (ChatItemId, Maybe Int64, Maybe ACIFileStatus, Maybe FilePath) -> (ChatItemId, Maybe CIFileInfo)
-toItemIdAndFileInfo' (chatItemId, fileId_, fileStatus_, filePath) =
-  case (fileId_, fileStatus_) of
-    (Just fileId, Just fileStatus) -> (chatItemId, Just CIFileInfo {fileId, fileStatus, filePath})
-    _ -> (chatItemId, Nothing)
+deleteGroupExpiredCIs :: DB.Connection -> User -> GroupInfo -> UTCTime -> UTCTime -> IO ()
+deleteGroupExpiredCIs db User {userId} GroupInfo {groupId} expirationDate createdAtCutoff = do
+  DB.execute db "DELETE FROM messages WHERE group_id = ? AND created_at <= ?" (groupId, expirationDate)
```

Comment:
maybe it's better to add this filter here as well..
```suggestion
  DB.execute db "DELETE FROM messages WHERE group_id = ? AND created_at <= ? AND created_at <= ?" (groupId, expirationDate, createdAtCutoff)
```

Reply:
why not `min` outside of query